Solutions/Microsoft 365/Analytic Rules/sharepoint_file_transfer_folders_above_threshold.yaml

```diff
@@ -0,0 +1,45 @@
+id: 8a547285-801c-4290-aa2e-5e7e20ca157d
+name: Office365 Sharepoint File transfer above threshold
+description: |
+  'Identifies Office365 Sharepoint File Transfers with distinct folder count above certain threshold in a 15min time period.
+   Please note that entity mapping for arrays is not supported, so when there is a single value in an array, we will pull that
+   value from the array as a single string to populate the entity to support entity mapping features within Sentinel. Additionally,
+   if the array is multivalued, we will input a string to indicate this with a unique hash so that matching will not occur.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: Office365
+    dataTypes:
+      - OfficeActivity
+queryFrequency: 15m
+queryPeriod: 15m
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Exfiltration
+relevantTechniques:
+  - T1020
+query: |
+  let threshold = 500;
```

Comment:
Hi, can you please fix the query here?
The error we are encountering is in KQL validation due to the fact that Kusto  doesn't allow us to use custom names in the summarize clause without wrapping them in aggregation functions like count() or make_set().
```
let threshold = 500;
OfficeActivity
| where EventSource == "SharePoint" and OfficeWorkload has_any("SharePoint", "OneDrive") and Operation has_any ("FileDownloaded", "FileSyncDownloadedFull", "FileSyncUploadedFull", "FileUploaded")
| summarize count_distinct_SourceRelativeUrl = count_distinct(SourceRelativeUrl), dirlist = make_set(SourceRelativeUrl) by UserId, ClientIP, UserAgent, bin(TimeGenerated, 15m)
| where count_distinct_SourceRelativeUrl >= threshold
| extend DirSample = iff(array_length(dirlist) == 1, tostring(dirlist[0]), strcat("SeeDirListField", "_", tostring(hash(tostring(dirlist))))

```

Reply:
I believe that if not setting name I was getting another error.
Only seeing this type of error
```
  Error Message:
   Template Id: 8b4f03e7-3460-4401-824d-e65a8dd464f0 is not valid in Line: 4 col: 43
                    Errors: The name 'count_distinct' does not refer to any known function., Code: 'KS211', Severity: 'Error', Location: '275..289'
```

count_distinct is valid - https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/count-distinct-aggfunction
if using count(), altering logic
without name, how do I check threshold?
and using in my prod deployment (deployed from ARM template)

Reply:
Hi juju4, I will get back to you by 27 Oct 2023 on this.

Reply:
Done